docs/hugo/content/design/ADR-2022-11-Change-Detection.md

```diff
@@ -26,24 +28,36 @@ Some fields are optional in the Swagger, but are required by Azure. In these cas
+Azure is not guaranteed to return the exact same resource that was PUT. 
+
+* Some services normalize identifier fields (converting them to all lowercase)
+* Some fields accept a wider range of values than they return.
 
-Potential mitigation: Generally speaking, Azure is expected to be "case-preserving, case-insensitive", so we should compare all string fields in a case insensitive manner. Region names and virtual machines SKUS are two known special cases where we may need to do some special case handling. For example, regions specified as `eastus` may be returned as `East US`.
+For example the CosmosDB API will return the region `West US` when the resource specified `westus`.
+
+Potential mitigations: 
+
+* Generally speaking, Azure identifiers are expected to be "case-preserving, case-insensitive", so we should compare those string fields in a case insensitive manner. 
+* Region names and virtual machines SKUS are two known special cases where we may need to do some special case handling. 
 
 ### Array ordering
 
-Some resources have arrays of sub-resources. These arrays may not be guaranteed to be returned in the same order as they were specified. 
+Some resources have arrays of items. These arrays may not be guaranteed to be returned in the same order as they were specified. 
 
-Potential mitigation: Where the items in an array have a known identifier (easy for nested resources), use that identifier to match them up. Where the items in an array do not have a known identifier, compare them by index. We may find need for 
+Potential mitigation: In the common case where the items in an array have a known identifier, use that identifier to match them up. Where the items in an array do not have a known identifier, compare them by index.
 
 ### Changes to test recordings
 
-When a resource changes from PUT only to GET+PUT, we'll need to re-record the test results for that resource. To avoid having to re-record all the tests in one go, we probably want to have some way to migrate to the new approach in a controlled fashion. 
+When a resource changes from PUT only to GET+PUT, we'll need to re-record the test results for that resource. To avoid having to re-record all the tests in one go, we probably want to have some way to migrate to the new approach in a controlled fashion.
 
 Potential mitigation: We could have a configuration flag that allows us to switch between the two approaches for a specific resource. We'd make GET+PUT the default, explicitly configure all our existing resources as PUT-only, and migrate in a controlled fashion, prioritizing those resources where we know we have the most issues with the PUT-only approach.
 
+This will also allow us to leave problematic resources a PUT only if we find that it's difficult or cumbersome to use GET+PUT because of issues doing the comparison.
```

Comment:
```suggestion
This will also allow us to leave problematic resources using only PUT if we find that it's difficult or cumbersome to use GET+PUT because of issues doing the comparison.
```